src/main/java/com/minecolonies/client/gui/WindowHutGuardTower.java

```diff
@@ -0,0 +1,467 @@
+
+    /**
+     * Patrol manually or automatically.
+     */
+    private boolean patrolManually = false;
+
+    /**
+     * The task of the guard, following the Task enum.
+     */
+    private BuildingGuardTower.Task task = BuildingGuardTower.Task.GUARD;
+
+    /**
+     * The job of the guard, following the GuarJob enum.
+     */
+    private BuildingGuardTower.GuardJob job = null;
+
+    /**
+     * The list of manual patrol targets.
+     */
+    private ArrayList<BlockPos> patrolTargets = new ArrayList<>();
+
+    /**
+     * The patrol list.
+     */
+    private ScrollingList patrolList;
+
+    /**
+     * Buttons used in the application:
+     */
+    private final Button buttonTaskPatrol;
+    private final Button buttonTaskFollow;
+    private final Button buttonTaskGuard;
+    private final Button buttonSetTarget;
+
+
+    /**
+     * Constructor for the window of the guardTower hut
+     *
+     * @param building {@link BuildingGuardTower.View}
+     */
+    public WindowHutGuardTower(BuildingGuardTower.View building)
+    {
+        super(building, Constants.MOD_ID + HUT_GUARD_TOWER_RESOURCE_SUFFIX);
+
+        pullInfoFromHut();
+
+        registerButton(BUTTON_JOB, this::switchJob);
+        registerButton(BUTTON_ASSIGNMENT_MODE, this::switchAssignmentMode);
+        registerButton(BUTTON_PATROL_MODE, this::switchPatrolMode);
+        registerButton(BUTTON_RETRIEVAL_MODE, this::switchRetrievalMode);
+
+        registerButton(BUTTON_TASK_PATROL, this::switchTask);
+        registerButton(BUTTON_TASK_FOLLOW, this::switchTask);
+        registerButton(BUTTON_TASK_GUARD, this::switchTask);
+        registerButton(BUTTON_SET_TARGET, this::setTarget);
+
+        buttonTaskPatrol = this.findPaneOfTypeByID(BUTTON_TASK_PATROL, Button.class);
+        buttonTaskFollow = this.findPaneOfTypeByID(BUTTON_TASK_FOLLOW, Button.class);
+        buttonTaskGuard = this.findPaneOfTypeByID(BUTTON_TASK_GUARD, Button.class);
+
+        buttonSetTarget = this.findPaneOfTypeByID(BUTTON_SET_TARGET, Button.class);
+        handleButtons();
+    }
+
+    /**
+     * Handle the task buttons correctly.
+     */
+    private void handleButtons()
+    {
+        Button buttonJob = this.findPaneOfTypeByID(BUTTON_JOB, Button.class);
+
+        if(job != null)
+        {
+            if (job.equals(BuildingGuardTower.GuardJob.KNIGHT))
+            {
+                buttonJob.setLabel(LanguageHandler.format("com.minecolonies.gui.workerHuts.knight"));
+            }
+            else
+            {
+                buttonJob.setLabel(LanguageHandler.format("com.minecolonies.gui.workerHuts.ranger"));
+            }
+        }
+
+        buttonJob.setEnabled(assignManually);
+
+        String auto = LanguageHandler.format("com.minecolonies.gui.workerHuts.modeA");
+        String manual = LanguageHandler.format("com.minecolonies.gui.workerHuts.modeM");
+
+        String on = LanguageHandler.format("com.minecolonies.gui.workerHuts.retrieveOn");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'on' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)